underfs/cosn/src/main/java/alluxio/underfs/cosn/CosNUnderFileSystemFactory.java

```diff
@@ -40,6 +42,20 @@ public boolean supportsPath(String path) {
 
   @Override
   public boolean supportsPath(String path, UnderFileSystemConfiguration conf) {
-    return supportsPath(path);
+    if (!supportsPath(path)) {
+      return false;
+    }
+    // if the user has explicitly specified a UFS version, do a strict match
+    if (conf.isSetByUser(PropertyKey.UNDERFS_VERSION)
```

Comment:
should this be a strict match or a prefix match?

Reply:
The prefix matching is done just before the version matching (`supportsPath(path)`); the strict match means the version also has to match.

It's better to be rephrased as "if the user has explicitly specified a UFS version, the version also has to match exactly, in addition to path prefix."